src/pages/api/contact-details/index.js

```diff
@@ -0,0 +1,46 @@
+import * as HttpStatus from 'http-status-codes'
+import {
+  authoriseServiceAPIRequest,
+  externalAPIRequest,
+} from '@/utils/serviceApiClient'
+
+export default authoriseServiceAPIRequest(async (req, res) => {
+  if (req.method !== 'DELETE' && req.method !== 'PATCH') return
```

Comment:
This can be refactored into a switch which would allow better extensibility later. This line can be a default case